### ISSUE TITLE
'Incomplete first view data for WPT test' while executing webpagetest scripts with few steps in it

### ISSUE THREAD

**Issue by serputko** [sitespeedio/sitespeed.io#1897]
Hi guys,

i'm trying to setup webpagetest scripts with sitespeedio and facing ERROR: Incomplete first view data for WPT test. It is reproducible both while using --webpagetest.script or --webpagetest.file 
webpagetest script is 
```
setEventName	Open main page
navigate	http://www.webpagetest.org/
setEventName	Open History
setValue	name=url	foo
execAndWait	document.querySelector("[title='Test History']").click();
```
So i run sitespeedio container with my wpt script:
`docker run --name sitespeed --volumes-from=jenkins --network=host --privileged=true --shm-size=1g --rm -v /var/lib/sitespeed.io/:/sitespeed.io sitespeedio/sitespeed.io --graphite.host localhost --graphite.port 2003 --graphite.httpPort 8080 --browsertime.browser chrome --browsertime.iterations 1 --webpagetest.host localhost --webpagetest.location Test --webpagetest.runs 1 --webpagetest.includeRepeatView true --webpagetest.script setEventName\tOpen main page\nnavigate\thttp://www.webpagetest.org/\nsetEventName\tOpen History\nsetValue\tname=url\tfoo\nexecAndWait\tdocument.querySelector("[title='Test History']").click(); http://www.google.com/`
and get next output:

> [2018-01-31 10:55:31] INFO: Versions OS: linux 4.4.0-1044-aws nodejs: v8.9.4 sitespeed.io: 6.3.0 browsertime: 2.1.4 coach: 1.2.0
> [2018-01-31 10:55:31] INFO: Sending url http://www.google.com/ to test on localhost
> [2018-01-31 10:55:31] INFO: Starting chrome for analysing http://www.google.com/ 1 time(s)
> [2018-01-31 10:55:31] INFO: Testing url http://www.google.com/ run 1
> [2018-01-31 10:55:47] INFO: 19 requests, 547.87 kb, backEndTime: 1.12s, firstPaint: 1.37s, firstVisualChange: 1.60s, DOMContentLoaded: 1.51s, Load: 2.55s, speedIndex: 1726, visualComplete85: 1.60s, lastVisualChange: 4.57s, rumSpeedIndex: 1372
> [2018-01-31 10:56:32] INFO: Got http://www.google.com/ analysed with id 180131_DJ_34536c998b0c88d3f10bca2869388930 from localhost
> [2018-01-31 10:56:32] ERROR: Incomplete first view data for WPT test 180131_DJ_34536c998b0c88d3f10bca2869388930, run 1
> [2018-01-31 10:56:32] ERROR: TypeError: Cannot convert undefined or null to object
>     at Function.keys (<anonymous>)
>     at metricPaths.reduce (/usr/src/app/lib/support/metricsFilter.js:65:30)
>     at Array.reduce (<anonymous>)
>     at Object.filterMetrics (/usr/src/app/lib/support/metricsFilter.js:35:24)
>     at Object.filterMessage (/usr/src/app/lib/support/filterRegistry.js:48:42)
>     at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:61:30)
>     at queue.process.message (/usr/src/app/lib/core/queueHandler.js:210:32)
>     at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
>     at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
>     at runCallback (timers.js:789:20)
>     at tryOnImmediate (timers.js:751:5)
>     at processImmediate [as _immediateCallback] (timers.js:722:5)
> [2018-01-31 10:56:35] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.google.com/2018-01-31-10-55-31
> [2018-01-31 10:56:35] INFO: Finished analysing http://www.google.com/

I checked webpagetest report. It is generated and has data for all steps from my script. 
Could you suggest some way how to fix that sitespeed error?

Also I noticed that when i leave just 1 step in script:

```
setEventName	Open main page
navigate	http://www.webpagetest.org/
```

that error is not displayed

i thought that maybe execAndWait command causes some problem but script with 2 'navigate' commands is not working too

Thanks in advance!

**Comment by serputko** [comment]
Hi @soulgalore, thanks for your answer.

It's okay if sitespeedio can't handle multiple pages results now. I can find necessary info about script execution in wpt report. But i don't want to see that error, it fails my CI builds :)

**Comment by soulgalore** [comment]
Did a quick look and it seems like the WebPageTest result lack breakdown (type/size) and that makes it fails. See if I can do a push tonight/tomorrow but I wonder if WPT should return it?

**Comment by serputko** [comment]
Are you talking about this http://prntscr.com/i85n31? It is present in wpt report in test where sitespeed throws error

**Comment by soulgalore** [comment]
Yep but it looked like it's missing in the JSON we got, but I'll look more into it, at least catch or prevent the error.

**Comment by serputko** [comment]
Thanks in advance

**Comment by soulgalore** [comment]
I've pushed a fix and can release it later today. However we have another check that if you configured to get the video and the SpeedIndex is 0 (sometimes WPT isn't working correctly and that the only way to know) and I also got that when I tested it.

**Comment by soulgalore** [comment]
@serputko released a fix that at release take care of the error. Let me know when you had time to try it out and hope it work as it should!

**Comment by serputko** [comment]
@soulgalore Thanks
Now it looks like this:
[2018-02-02 11:06:49] ERROR: Incomplete first view data for WPT test 180202_A8_b2366c1d461578d8b278c9aae5214576, run 1
[2018-02-02 11:06:49] ERROR: Metricsfilter: The breakdown is missing from the metrics breakdown.*.requests
[2018-02-02 11:06:49] ERROR: Metricsfilter: The breakdown is missing from the metrics breakdown.*.requests
[2018-02-02 11:06:49] ERROR: Metricsfilter: The breakdown is missing from the metrics breakdown.*.bytes
[2018-02-02 11:06:49] ERROR: Metricsfilter: The breakdown is missing from the metrics breakdown.*.bytes

Think this issue should be closed, i will create an issue for webpagetest.

**Comment by soulgalore** [comment]
I think that could be because of the SpeedIndex missing but video was configured. Do you mean it misses at all or for parts of your run?

**Comment by serputko** [comment]
Ahh, i found the reason. Link to wpt report in displayed on Page summary, but if select run # summary some required data is missing.

Think this issue should be closed. Thanks 👍

**Comment by chrischjh** [comment]
Hi @serputko , can you share more detail on how to make it correct? I got the same error. thanks a lot

**Comment by quicoto** [comment]
Hello,

I'm getting the same error using 7.7.3

**Comment by soulgalore** [comment]
Hi @quicoto could you please share what we can do to reproduce, see https://www.sitespeed.io/documentation/sitespeed.io/bug-report/.

Thanks!
Peter

**Comment by quicoto** [comment]
@soulgalore thanks for the prompt reply. Do let me know if you want me to open a new issue, since this one is closed.

Using with CLI 

"sitespeed.io": "^7.7.3"

Command

```
sitespeed.io https://my-secret-url --preScript UAT-login.js -n 1  --webpagetest.key MY_SECRET_KEY
```

The pre-script simply logs me in into the application.

It works fine without the webpagetest plugin. Now, when I add the key, I get:

```
INFO: Got https://my-secret-url analysed with id 190118_EW_617662db12ece50446a45d9f811fa84b from https://www.webpagetest.org
[2019-01-18 10:51:10] ERROR: Incomplete first view data for WPT test 190118_EW_617662db12ece50446a45d9f811fa84b, run 1
```

Thank you in advance.

**Comment by soulgalore** [comment]
Hi @quicoto if you look at the webpagetest server, does it look ok there? If you turn on —verbose you can see the json back from webpagetest and see whats missing. It’s hard for me to guess :)

Best
Peter

**Comment by chrischjh** [comment]
In my case, it was due to ffmpeg not the latest version, thus SpeedIndex was not generated. I followed the wpt agent installation advise, replaced with latest version, fixed.

“ImageMagick includes ffmpeg in the path automatically but not the latest version. It is usually easiest to just copy the static ffmpeg build over the one installed by ImageMagick”

https://github.com/WPO-Foundation/wptagent/blob/master/docs/install.md
